lib/parser/builders/default.rb

```diff
@@ -359,18 +359,23 @@ def dedent_string(node, dedent_level)
       if !dedent_level.nil?
         dedenter = Lexer::Dedenter.new(dedent_level)
 
-        if node.type == :str
+        case node.type
+        when :str
           str = node.children.first
           dedenter.dedent(str)
-        elsif node.type == :dstr || node.type == :xstr
-          node.children.each do |str_node|
+        when :dstr, :xstr
```

Comment:
This is so nice for me in the unparser case as I now can potentially collapse the xstr and dstr handling against each other.